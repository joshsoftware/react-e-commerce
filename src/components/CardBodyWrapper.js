import React from 'react';
import { CardBody } from 'reactstrap';
import { PropTypes } from 'prop-types';

const CardBodyWrapper = ({ body_content, className }) => {
  return <CardBody className={className}>{body_content}</CardBody>;
};

export default CardBodyWrapper;

CardBodyWrapper.propTypes = {
  body_content: PropTypes.array.isRequired,
  className: PropTypes.string
};
