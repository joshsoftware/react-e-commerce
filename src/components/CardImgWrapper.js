import React from 'react';
import { CardImg } from 'reactstrap';
import { PropTypes } from 'prop-types';
const CardImgWrapper = ({ width, src, top, bottom, className, tag }) => {
  return (
    <CardImg
      top={top}
      bottom={bottom}
      tag={tag}
      width={width}
      className={className}
      src={src}
      alt={'Product Image'}
    />
  );
};
export default CardImgWrapper;

CardImgWrapper.propTypes = {
  width: PropTypes.string,
  src: PropTypes.string.isRequired,
  tag: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
  className: PropTypes.string,
  top: PropTypes.bool,
  bottom: PropTypes.bool
};
