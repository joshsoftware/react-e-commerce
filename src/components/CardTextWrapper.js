import React from 'react';
import { CardText } from 'reactstrap';
import { PropTypes } from 'prop-types';

const CardTextWrapper = ({ text, className }) => {
  return <CardText className={className}>{text}</CardText>;
};

export default CardTextWrapper;

CardTextWrapper.propTypes = {
  text: PropTypes.string.isRequired,
  className: PropTypes.string
};
