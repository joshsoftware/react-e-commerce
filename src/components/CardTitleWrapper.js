import React from 'react';
import { CardTitle } from 'reactstrap';
import { PropTypes } from 'prop-types';

const CardTitleWrapper = ({ title, className }) => {
  return <CardTitle className={className}>{title}</CardTitle>;
};

export default CardTitleWrapper;

CardTitleWrapper.propTypes = {
  title: PropTypes.string.isRequired,
  className: PropTypes.string
};
