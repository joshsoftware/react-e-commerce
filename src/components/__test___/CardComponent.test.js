import React from 'react';
import { render } from '@testing-library/react';
import { Card } from 'reactstrap';
import CardImgWrapper from '../CardImgWrapper';
import CardTitleWrapper from '../CardTitleWrapper';
import CardTextWrapper from '../CardTextWrapper';
import CardBodyWrapper from '../CardBodyWrapper';

describe('Product Card', () => {
  it('Must Return Product Card', () => {
    let body_content = [];
    let i = 0;
    body_content.push(<CardTitleWrapper key={i++} title={'Project Title'} />);
    body_content.push(<CardTextWrapper key={i++} text={'$1000'} />);

    const { asFragment } = render(
      <Card>
        <CardImgWrapper
          src={
            'https://assets.ajio.com/medias/sys_master/root/hb7/h8b/15083561943070/-473Wx593H-460422382-black-MODEL.jpg'
          }
        />
        <CardBodyWrapper body_content={body_content} />
      </Card>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
