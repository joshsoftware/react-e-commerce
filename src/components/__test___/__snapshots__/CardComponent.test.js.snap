// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Card Must Return Product Card 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <img
      alt="Product Image"
      class="card-img"
      src="https://assets.ajio.com/medias/sys_master/root/hb7/h8b/15083561943070/-473Wx593H-460422382-black-MODEL.jpg"
    />
    <div
      class="card-body"
    >
      <div
        class="card-title"
      >
        Project Title
      </div>
      <p
        class="card-text"
      >
        $1000
      </p>
    </div>
  </div>
</DocumentFragment>
`;
